packages/splitter/src/useSplitter.ts

```diff
@@ -54,45 +65,48 @@ export interface IUseSplitterProps extends Omit<HTMLProps<any>, 'onChange'> {
   /** A callback that receives updated valueNow when a separator moves */
   onChange?: (valueNow: number) => void;
   /** A boolean that changes splitter from left-to-right to right-to-left mode */
-  rtl?: boolean;
+  placement: SplitterPlacement;
 }
 
 export interface IUseSplitterReturnValue {
   getSeparatorProps: <T>(options?: T & HTMLProps<any>) => any;
+  getPrimaryPaneProps: <T>(options?: T & HTMLProps<any>) => any;
 }
 
-// Pointer coordinates do not account padding, margins, or height/width of separator
-// This function takes mouse or touch value minus padding or margin position minus the width or height of the separator divided by 2
-// This aligns the pointer and the separator correctly
-// When passed a viewportWidth, subtract pointer position and offsets to invert the calculation for RTL mode.
-export const calculateOffset = (
+// takes pointer distance from edge of viewport to the pointer
+// minus the offset of pointer position to the separator in the DOM layout
+// minus the separator width or height divided by 2 to place cursor in middle of separator
+export const normalizePointerToSeparator = (
+  bodyPadding: number,
   pointerPosition: number,
-  paddingOrMarginPosition = 0,
-  offsetDimension = 0,
-  viewportWidth = 0
+  separatorHeightOrWidth = 0,
+  viewportWidthOrHeight = 0
 ) => {
-  if (viewportWidth === 0) {
-    return pointerPosition - paddingOrMarginPosition - Math.ceil(offsetDimension / 2);
+  // if placement end use the pointerPosition relative to the left of the viewport
+  if (viewportWidthOrHeight === 0) {
+    return pointerPosition - bodyPadding - Math.floor(separatorHeightOrWidth / 2);
   }
 
-  return viewportWidth - pointerPosition - paddingOrMarginPosition - offsetDimension;
+  // if placement start we need to invert the pointerPosition so its relative to right of the viewport
+  return (
+    viewportWidthOrHeight - pointerPosition - bodyPadding - Math.floor(separatorHeightOrWidth / 2)
+  );
```

Comment:
Refactored `calculateOffset` function to `normalizePointerToSeparator` for more clarity